pkg/metrics/metrics.go

```diff
@@ -132,15 +137,46 @@ var (
 			Help: "Monotonic count of CSV upgrades",
 		},
 	)
+
+	CSVSyncCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "csv_sync_total",
+			Help: "Monotonic count of CSV syncs",
+		},
+		[]string{PHASE_LABEL, NAME_LABEL},
+	)
+
+	SubscriptionSyncCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "subscription_sync_total",
+			Help: "Monotonic count of subscription syncs",
+		},
+		[]string{NAME_LABEL},
+	)
 )
 
 func RegisterOLM() {
 	prometheus.MustRegister(csvCount)
 	prometheus.MustRegister(CSVUpgradeCount)
+	prometheus.MustRegister(CSVSyncCount)
 }
 
 func RegisterCatalog() {
 	prometheus.MustRegister(installPlanCount)
 	prometheus.MustRegister(subscriptionCount)
 	prometheus.MustRegister(catalogSourceCount)
+	prometheus.MustRegister(SubscriptionSyncCount)
+}
+
+func CounterForCSV(name, phase string) (prometheus.Counter, error) {
+	return CSVSyncCount.GetMetricWith(map[string]string{
```

Comment:
I personally find `.WithLabelValues` more ergonomic to use for this https://github.com/prometheus/client_golang/blob/ff1d4e21c12e4c0b190a166d96507786c8f989d6/prometheus/counter.go#L203

Reply:
I don't really want OLM to panic if this fails though, right?

Reply:
if you pass the wrong labels your instrumentation is still going to be wrong, but it's a style choice, you're free to choose yours :) we prefer `.WithLabelValues` generally